src/joint_state_listener.cpp

```diff
@@ -86,7 +86,11 @@ void JointStateListener::callbackFixedJoint(const ros::TimerEvent& e)
 void JointStateListener::callbackJointState(const JointStateConstPtr& state)
 {
   if (state->name.size() != state->position.size()){
-    ROS_ERROR("Robot state publisher received an invalid joint state vector");
+    if (state->position.empty()){
+      ROS_WARN_THROTTLE(300, "Robot state publisher ignored a JointState message whose position member was empty");
```

Comment:
It would be great to list the names, and also mention that this message will not reappear for 300 seconds. For those who don't know that this is a throttled output.